### PR TITLE
Use native Array.prototype.findIndex function

### PR DESCRIPTION
**Pull request by fredj**
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

### CODE REVIEW

File: closure/goog/array/array.js [hunk 1]
```diff
@@ -501,7 +501,7 @@ goog.array.count = function(arr, f, opt_obj) {
  * return that element.
  * @param {Array<T>|goog.array.ArrayLike} arr Array or array
  *     like object over which to iterate.
- * @param {?function(this:S, T, number, ?) : boolean} f The function to call
+ * @param {!function(this:S, T, number, ?) : boolean} f The function to call
```
Comment (joeltine) [inline closure/goog/array/array.js]: While correct, these type changes have the potential to break a bunch of ppl.

**Comment by joeltine** [comment]
There are some compilation errors in the travis report.

**Comment by fredj** [comment]
This pull request depends on https://github.com/google/closure-compiler/pull/1190

**Comment by concavelenz** [comment]
Generally, I would rather keep adding these unless there is some inherit performance advantage to the native methods (which there generally is not as the VMs just write these methods in JS).

**Comment by joeltine** [comment]
@concavelenz did you mean "rather not"?

**Comment by concavelenz** [comment]
sorry, yes "rather not".   Is there a clear performance advantage?

**Comment by joeltine** [comment]
@fredj Got a jsperf?

**Comment by fredj** [comment]
I don't need this anymore, closing

**Pull request closed without merging**